cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java

```diff
@@ -200,6 +214,22 @@ default CloseableIterator<StructuredRow> scan(Range keyRange, int limit, String
   CloseableIterator<StructuredRow> multiScan(Collection<Range> keyRanges,
                                              int limit) throws InvalidFieldException, IOException;
 
+  /**
+   * Scan a collection of rows from the table matching the set of partial primary keys.
+   * The rows returned will be sorted on the primary key order.
+   *
+   * @param partialKeysCollections collection of key ranges for the scan
+   * @param limit maximum number of rows to return
+   * @return a {@link CloseableIterator} of rows
+   * @throws InvalidFieldException if any of the keys are not part of the table schema, or the types of the value
+   *                               do not match
+   * @throws IOException if there is an error scanning the table
+   */
+  CloseableIterator<StructuredRow> multiScanPartialKeys(
+    Collection<? extends Collection<Field<?>>> partialKeysCollections,
```

Comment:
Similar to above, can it be handled by Collection of singleton ranges?

Reply:
Similar reasoning above